src/compressor/zlib/CompressionZlib.cc

```diff
@@ -134,7 +134,7 @@ int CompressionZlib::decompress(const bufferlist &in, bufferlist &out)
       strm.avail_out = max_len;
       strm.next_out = c_out;
       ret = inflate(&strm, Z_NO_FLUSH);
-      if (ret != Z_OK && ret != Z_STREAM_END) {
```

Comment:
this worry me, from http://www.zlib.net/zlib_how.html it looks like we need to exit if meeting Z_BUF_ERROR.